### PR TITLE
PLNSRVCE-587: Add ninja-build package required by netty-tcnative

### PR DESCRIPTION
**Pull request by dwalluck**

### CODE REVIEW

**Comment by gabemontero** [comment]
spinning pac jobs are complete:

```
gmontero ~ $ oc get pr | grep vf
jvm-build-hacbs-jdk17-builder-on-pull-request-vflqh   True        Completed   3h26m       3h14m
gmontero ~ $ oc get pr | grep gm6
jvm-build-hacbs-jdk8-builder-on-pull-request-gm6tk    True        Completed   3h26m       3h12m
gmontero ~ $ 
```

the e2e failure is not a flake per say, the last e2e check failed

/test jvm-build-service-in-repo-e2e

see if it fails again

**Comment by gabemontero** [comment]
so the test pased this time

` INFO[2022-08-16T23:28:52Z] Step jvm-build-service-in-repo-e2e-jvm-build-service-e2e-in-repo-e2e succeeded after 22m20s. `

```
    types.go:25: time: 2022-08-16 23:28:48.903087458 +0000 UTC m=+870.259125264: contaminated dependencybuild: ff82ef848e8068f617f471fb3db4e1ac
    types.go:25: time: 2022-08-16 23:28:48.970694342 +0000 UTC m=+870.326732148: number of artifactbuilds: 4
    types.go:25: time: 2022-08-16 23:28:49.035748715 +0000 UTC m=+870.391786511: simple-jdk8 State: ArtifactBuildComplete
    types.go:25: time: 2022-08-16 23:28:49.100822559 +0000 UTC m=+870.456860355: shaded-jdk11 State: DependencyBuildStateComplete
--- PASS: TestExampleRun (870.44s)
    --- PASS: TestExampleRun/pipelinerun_completes_successfully (0.08s)
    --- PASS: TestExampleRun/artifactbuilds_and_dependencybuilds_generated (210.33s)
    --- PASS: TestExampleRun/some_artfactbuilds_and_dependencybuilds_complete (630.29s)
    --- PASS: TestExampleRun/contaminated_build_is_resolved (0.27s)
PASS
ok  	github.com/redhat-appstudio/jvm-build-service/openshift-with-appstudio-test/e2e	870.461s
```

but the must gather failed ... saw the same thing in  my current PR

probably some CI flakiness, let's retest tomorrow

File: builder-images/hacbs-jdk11-builder/Dockerfile
Comment (stuartwdouglas) [inline builder-images/hacbs-jdk11-builder/Dockerfile:34]: If we are just wget'ing something we should probably check the hash is as expected.

I do wonder a bit how far we can go with native packages, at some point it may be we just have to leave it to PNC for now.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

File: builder-images/hacbs-jdk11-builder/Dockerfile
Comment (dwalluck) [reply builder-images/hacbs-jdk11-builder/Dockerfile:34]: This makes the assumption that the GPG key is valid and checks the GPG signature to verify the package.

Comment (dwalluck) [reply builder-images/hacbs-jdk11-builder/Dockerfile:34]: Specifically this was required to build BoringSSL.

File: builder-images/Dockerfile.template
Comment (dwalluck) [inline builder-images/Dockerfile.template:34]: Import GPG key and check package name (which contains fingerprint)

File: builder-images/Dockerfile.template
Comment (dwalluck) [inline builder-images/Dockerfile.template:36]: This checks the package signature.

Comment (stuartwdouglas) [reply builder-images/Dockerfile.template:36]: Ah, sorry, my bad

**Review by stuartwdouglas — APPROVED** [review]

File: builder-images/Dockerfile.template
Comment (dwalluck) [reply builder-images/Dockerfile.template:36]: Yeah, it was using GPG signatures, but I think I still needed to verify the fingerprint of the downloaded GPG key, so thanks.

**Comment by gabemontero** [comment]
as we do not have tests quite yet in jvm-build-service-e2e (though that may change by as soon as Thursday/tomorrow) this technically passed our current tests, but let's see if we can get all green before merging.

/test jvm-build-service-e2e

**Comment by gabemontero** [comment]
all green

**Pull request merged**
